source/common/common/utility.h

```diff
@@ -297,6 +297,14 @@ class StringUtil {
    */
   static std::string join(const std::vector<std::string>& source, const std::string& delimiter);
 
+  /**
+   * Join elements of a sorted set into a string delimited by delimiter.
+   * @param source supplies the strings to join.
+   * @param delimiter supplies the delimiter to join them together.
+   * @return string combining elements of `source` with `delimiter` in between each element.
+   */
+  static std::string join(const std::set<std::string>& source, const std::string& delimiter);
```

Comment:
Can we maybe just have a templatized `join` that works on iterators? That could replace the one for vector as well.

Reply:
Yeah, I was wondering if there might be something like that, but.... seemed like a lot of trouble! This duplicated join() broke some other tests I didn't notice, so I have already addressed that by just inlining the single use of join<set>(). Does that sound ok?

Reply:
It's OK, but it's probably only a couple of lines to make it a template with iterators :)